crates/consensus/common/src/validation.rs

```diff
@@ -280,27 +317,16 @@ pub fn validate_header_regarding_parent(
     // TODO Check difficulty increment between parent and child
     // Ace age did increment it by some formula that we need to follow.
 
-    let mut parent_gas_limit = parent.gas_limit;
-
-    // By consensus, gas_limit is multiplied by elasticity (*2) on
-    // on exact block that hardfork happens.
-    if chain_spec.fork(Hardfork::London).transitions_at_block(child.number) {
-        parent_gas_limit = parent.gas_limit * chain_spec.base_fee_params.elasticity_multiplier;
-    }
-
-    // Check gas limit, max diff between child/parent gas_limit should be  max_diff=parent_gas/1024
-    if child.gas_limit > parent_gas_limit {
-        if child.gas_limit - parent_gas_limit >= parent_gas_limit / 1024 {
-            return Err(ConsensusError::GasLimitInvalidIncrease {
-                parent_gas_limit,
-                child_gas_limit: child.gas_limit,
-            })
+    cfg_if::cfg_if! {
+        if #[cfg(feature = "optimism")] {
+            // On Optimism, the gas limit can adjust instantly, so we skip this check
+            // if the optimism feature is enabled in the chain spec.
```

Comment:
Believe the best explanation for this is in the [Optimism transaction fee docs](https://community.optimism.io/docs/developers/build/transaction-fees/#base-fee), specifically the base fee since it varies so highly.